plugins/lime-plugin-locate/src/communityLayer.js

```diff
@@ -0,0 +1,75 @@
+const isObject = (obj) => typeof obj === 'object' && obj !== null;
+
+const sameLocation = (coord1, coord2) => Number(coord1.lon) === Number(coord2.lon) && Number(coord1.lat) === Number(coord2.lat);
+
+const removeDuplicates = (collection, getIdentifier) => {
+    const identifierState = {};
+    return collection.filter((value) => {
+        const identifier = String(getIdentifier(value));
+        if (identifierState[identifier]) {
+            return false;
+        }
+        identifierState[identifier] = true;
+        return true;
+    });
+};
+
+const coordsToPoint = ({ coordinates, hostname }) => ({ type: 'Feature', geometry: { type: 'Point', coordinates: [Number(coordinates.lon), Number(coordinates.lat)] }, properties: { name: hostname } });
+
+const coordspairToLineString = (coordinates) => ({ type: 'Feature', geometry: { type: 'LineString', coordinates } });
+
+export function getCommunityLayer(hostname, stationLocation, nodeshash) {
+    //Run only if leaflet is loaded
```

Comment:
Add check and exception if leaflet is not loaded instead of comment

Reply:
In the new version this only builds the geoJSON so it doesn't need leaflet. Thankyou!